packages/python/plotly/plotly/express/_core.py

```diff
@@ -241,18 +241,26 @@ def make_trace_kwargs(args, trace_spec, trace_data, mapping_labels, sizeref):
                     sorted_trace_data = trace_data.sort_values(by=args["x"])
                     y = sorted_trace_data[args["y"]]
                     x = sorted_trace_data[args["x"]]
-                    trace_patch["x"] = x
+                    # trace_patch["x"] = x
 
                     if x.dtype.type == np.datetime64:
                         x = x.astype(int) / 10 ** 9  # convert to unix epoch seconds
 
                     if attr_value == "lowess":
-                        trendline = sm.nonparametric.lowess(y, x)
+                        # missing ='drop' is the default value for lowess but not for OLS (None)
+                        # we force it here in case statsmodels change their defaults
+                        trendline = sm.nonparametric.lowess(y, x, missing="drop")
+                        trace_patch["x"] = trendline[:, 0]
```

Comment:
why do we not need to filter the x data here?

Reply:
Oh, I see, LOWESS will just pass those through in the estimates... hmm that means that with NaNs there are gaps in the line. OK.

Reply:
so `missing="drop"` means "carry NaNs through" in LOWESS?

Reply:
So for lowess
```
missing : str
    Available options are 'none', 'drop', and 'raise'. If 'none', no nan
    checking is done. If 'drop', any observations with nans are dropped.
    If 'raise', an error is raised. Default is 'drop'.
```
nans are just removed, and 'drop' is the default value (so I did not change anything for this function call for lowess, I just made the option explicit in case `sm` changes the default, since I was quite surprised that the default was different for `ols`). But I had to make sure we take the `trace_patch["x"]` from the `sm` returned object where NaNs have been dropped, because we used to take the original x-data which could have NaNs and then x and y did not have the same length.

For ols
```
missing : str
    Available options are 'none', 'drop', and 'raise'. If 'none', no nan
    checking is done. If 'drop', any observations with nans are dropped.
    If 'raise', an error is raised. Default is 'none'.
```
so I chose `drop`. It's weird that it's not the same default... (statsmodels version 0.11.1).

Does this answer your question? No we're not carrying NaNs (as we were previously for ols) through but I don't think we should.

Reply:
yep, all clear now!